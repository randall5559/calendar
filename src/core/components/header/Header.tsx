import * as React from 'react';
import './header.css';

export const Header = (props) => (
  <header>
    <div className="logo">
      <img src={'../assets/clipboard.svg'} />
      <h1 >Cal</h1>
    </div>
    <div className="toolbar">
      <h2 className="current-month">{props.date}</h2>
    </div>
    <div className="date-change">
      <span>
        <button onClick={() => {
          if (props.month !== 0) {
            props.setToPreviousCalendarView({ month: props.month - 1, year: props.year });
          } else if (props.month === 0) {
            props.setToPreviousCalendarView({ month: 11, year: props.year - 1 })
          }

        }}>
          <img src={'../assets/chevron-sign-to-left.svg'} />
        </button>
      </span>
      <span>
        <button onClick={() => {
          if (props.month !== 11) {
            props.setToNextCalendarView({ month: props.month + 1, year: props.year })
          } else if (props.month === 11) {
            props.setToNextCalendarView({ month: 0, year: props.year + 1 })
          }
        }}>
          <img src={'../assets/chevron-sign-to-right.svg'} />
        </button>
      </span>
    </div>
  </header>
);
