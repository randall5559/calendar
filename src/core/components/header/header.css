header {
    display: flex;
    flex: 1 100%;
    padding: 10px;
    height: 50px;
    flex-direction: row;
    align-items: center;
    justify-content: center;
}

img {
    display: inline-block;
    padding: 10px;
    width: 30px;
    height: 30px
}

h1 {
    display: inline-block;
    margin: 0;
    color: #4A4A4A;
    font-weight: 400;
    font-family: Arial, Helvetica, sans-serif;
}

.logo {
    flex: 1;
    display: flex;
    align-items: center;
}

.toolbar{
    position: absolute;
}

button {
    background-color: transparent;
    border: none;
    cursor: pointer;
}
