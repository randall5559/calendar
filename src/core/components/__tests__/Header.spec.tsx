import * as React from "react";
import { shallow } from "enzyme";

import { Header } from '../header/Header';

describe('Header Component', () => {
  it('should render correctly', () => {
    const tree = shallow(
      <Header />
    );
    expect(tree).toMatchSnapshot();
  });
});
