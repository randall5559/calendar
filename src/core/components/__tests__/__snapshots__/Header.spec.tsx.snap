// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <img
          src="../assets/clipboard.svg"
        />,
        <h1>
          Things To Do
        </h1>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "src": "../assets/clipboard.svg",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Things To Do",
        },
        "ref": null,
        "rendered": "Things To Do",
        "type": "h1",
      },
    ],
    "type": "header",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            src="../assets/clipboard.svg"
          />,
          <h1>
            Things To Do
          </h1>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "src": "../assets/clipboard.svg",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Things To Do",
          },
          "ref": null,
          "rendered": "Things To Do",
          "type": "h1",
        },
      ],
      "type": "header",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
